reference/src/main/java/oshi/api/OSHI.java

```diff
@@ -0,0 +1,78 @@
+package oshi.api;
+
+import com.sun.jna.Platform;
+
+import oshi.driver.SystemDriver;
+import oshi.driver.SystemDriverFreeBsd;
+import oshi.driver.SystemDriverLinux;
+import oshi.driver.SystemDriverMac;
+import oshi.driver.SystemDriverSolaris;
+import oshi.driver.SystemDriverWindows;
+
+public final class OSHI {
+
+	private static final PlatformEnum platform;
+
+	static {
+		if (Platform.isWindows()) {
+			platform = PlatformEnum.WINDOWS;
+		} else if (Platform.isLinux()) {
+			platform = PlatformEnum.LINUX;
+		} else if (Platform.isMac()) {
+			platform = PlatformEnum.MACOSX;
+		} else if (Platform.isSolaris()) {
+			platform = PlatformEnum.SOLARIS;
+		} else if (Platform.isFreeBSD()) {
+			platform = PlatformEnum.FREEBSD;
+		} else {
+			platform = PlatformEnum.UNKNOWN;
+			throw new RuntimeException("Unsupported platform");
+		}
+	}
+
+	/**
+	 * Get the system's platform type.
+	 * 
+	 * @return The platform type
+	 */
+	public static PlatformEnum getPlatform() {
+		return platform;
+	}
+
+	public static MultiSystem getSystem() {
+		if (platform == PlatformEnum.UNKNOWN)
+			throw new UnsupportedOperationException();
+		return new SystemDriver(platform);
+	}
+
+	public static WindowsSystem getWindowsSystem() {
+		if (platform != PlatformEnum.WINDOWS)
+			throw new UnsupportedOperationException();
+		return new SystemDriverWindows();
```

Comment:
These methods are like convenience methods for the cast as you described. The majority of users will still be calling `getSystem()`. I think it's nicer to have these optional methods rather than telling users to cast the return from `getSystem()` to get platform-specific information. We can document this intent on each method or, if you prefer, we can only provide `getSystem()`.

Reply:
As a cross-platform library we shouldn't make non-cross-platform code convenient. :) I would venture to say that class-specific access is an implementation detail in most cases.  What use case do you envision for this?

Reply:
The use case is getting a reference to a container that can give me platform-specific features:
```
WindowsSystem system = (WindowsSystem) OSHI.getSystem();
system.getFirmware().getWinOnlyAttr();
```
vs
```
WindowsSystem system = OSHI.getWindowsSystem();
system.getFirmware().getWinOnlyAttr();
```
In both cases I already know I'm on Windows. If I'm wrong, the first case throws `ClassCastException` and the second throws `UnsupportedOperationException`. The main difference is that the user needs to know about the `WindowsSystem` interface to make the cast beforehand. I think the name `getWindowsSystem` is sufficient warning that the user is leaving cross-platform territory.

Reply:
OK.  I guess I'm stuck in the OSHI 4 mindset of one interface to rule them all, but I can see benefit in making sure you're on a Windows system before getting services, etc.